### ISSUE TITLE
Support globs

### ISSUE THREAD

**Issue by The-Compiler** [snoack/flake8-per-file-ignores#1]
Hey, coming from flake8-putty, and this finally gives me hope to upgrade to flake8 3.0! 🎉 

However, I used to do this in flake8-putty, as I want to be more forgiving for docstrings in test files: `tests/*/test_*.py : +D100,D101,D401`

It'd be nice if I could do something like that with `flake8-per-file-ignores` too!

**Comment by snoack** [comment]
There you go. It made things somewhat more complicated though, because of `X100`:

> If an error to be ignored, no longer occurs for a given file, this will cause an `X100` error, in order to make you progressively reduce the number of ignores as legacy code gets rewritten or removed.

If we would just expand the filename with `glob.glob()`, and treat each resulting file individually, you would get an `X100` for each ignored error that wasn't reported for **all** files matching the glob pattern. But instead we only want an `X100` if an ignored error wasn't reported for **any** file matching the glob pattern.

Therefore, we have to process all files that match the same pattern in one step. This also means that we can no longer just patch the function that is called for each file, but need to hijack a more high-level API. However, as  a side effect, it now also works with flake8 3.0.0 (previously we required >=3.2.1).

I just released version 0.2 on PyPi, which includes this change.

**Comment by The-Compiler** [comment]
Hey! Thanks a lot for the quick fix, and apologies for not coming back to this earlier - I got swamped in stuff.

Right now I still have less time than I'd like for the project I need this in, so right now I'm still using flake8 2.6. However, I opened https://github.com/qutebrowser/qutebrowser/issues/3320 now so I don't forget about it. I'll let you know if something else comes up, but I think this change is what'll finally make it possible for me to make the switch! 🎉
